### DISCUSSION TITLE
How to stop extracted system includes when using `--query-driver`

### DISCUSSION THREAD

**Discussion by kammce** [clangd/clangd#2266] [Q&A] [answered]
I'm using Mac M1 OS 15.2. 

Clangd seems to work well when I'm using `--query-driver="**"` and `arm-none-eabi-g++` as the compiler. But when I use clang++ 17 I get errors. I ran the compilation command that comes from clangd and I discovered that this include is breaking my clangd build: `-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include` which comes from the extraction. 

How can I remove the system include compilation while still being able to use `--query-driver` because I need that for `arm-none-eabi-g++` to work. In my use case, it would be useful to keep `--query-driver="**"` so I can use clangd easily across workspace and across different compilers.

**Comment by HighCommander4** [comment]
If I'm understanding correctly that you want to use the system include extractor for g++ but not for clang++, you could try something like `--query-driver="**/*g++"`.

**Reply by kammce** [reply]
I've used this before and it didn't work. Now I figured out why this doesn't work 😆 Its kind of hilarious. `--query-driver="**/*g++"` cannot work because the end of clang++ is `g++`. So this matches `clang++` as well as `g++`.

**Reply by HighCommander4** [reply, marked as answer]
Oh, whoops! :laughing: 

Ok, here is a better suggestion. You can have multiple comma-separated patterns in `--query-driver`, so you could do:

```
--query-driver="**/g++,**/*-g++"
```

The first pattern, `**/g++`, will match unprefixed binaries like `/usr/bin/g++`.

The second pattern, `**/*-g++`, will match binaries with a prefix separated by a dash, such as `/path/to/arm-none-eabi-g++`.

Neither will match `clang++`.

**Reply by kammce** [reply]
Yup that works for me. Thank you for the response. I went with `--query-driver="**/g++,**/*-g++"` when I use the command line and `--query-driver=**/g++,**/*-g++` (without quotes) for vscode and that solved the problem for me. Thanks again!

**Reply by kammce** [reply]
Also, thank you for the quick response!
